fs/operations/operations.go

```diff
@@ -423,7 +423,14 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 				})
 			}
 			if doMultiThreadCopy(ctx, f, src) {
-				dst, err = multiThreadCopy(ctx, f, remotePartial, src, ci.MultiThreadStreams, tr)
+				options := []fs.OpenOption{hashOption}
```

Comment:
I think we should rename this `uploadOptions` and and move it and the following 6 lines up before the `if`, then we can re-use `uploadOptions` a bit further down when calling `Update` and `Put`.

I have on my TODO list to refactor `operations.Copy` - it is getting out of hand!

Reply:
hi, @ncw , thanks for the review! I updated the PR